internal/component/otelcol/exporter/kafka/kafka.go

```diff
@@ -27,11 +27,22 @@ func init() {
 
 		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
 			fact := kafkaexporter.NewFactory()
-			return exporter.New(opts, fact, args.(Arguments), exporter.TypeSignalConstFunc(exporter.TypeAll))
+			return exporter.New(opts, fact, args.(Arguments), GetSignalType)
```

Comment:
Would this still work if the config is reloaded to work with a different signal type?

Reply:
Yes: https://github.com/grafana/alloy/blob/2a991573b129157960f7c2fc3fa6600620803f02/internal/component/otelcol/exporter/exporter.go#L214